### DISCUSSION TITLE
Adding the RS232 Waveshare to acces Pylontech us5000

### DISCUSSION THREAD

**Discussion by mrpp** [alienatedsec/solis-ha-modbus-cloud#38] [General]
Hi, Trying to add the rs232 module to the solis setup i have from following your instructions. Once built and the client set up again as per instructions, is there anything else i need to connect battery view over network to batteries?

**Comment by alienatedsec** [comment]
RS232 is not part of my description? What are you trying to achieve?

**Reply by mrpp** [reply]
![alienatedsec-solis-ha-modbus-cloud-This-repo-is-a-workaround-for-Solis-inverters-to-connect-Solis-Cloud-and-the-local-Home-Assistant-regardless-of-the-model-of-your-Solis-datalogger-09-23-2024_01_21_PM](https://github.com/user-attachments/assets/bca6b289-f664-41d1-a8a9-534dbffd9f64)
Trying this, but with rs232 for the communication to pylontech batteries. I have everything else set and working for the solis side of things.

**Comment by mrpp** [comment]
So rather than connect direct from PC to battery console connection with a PYLONTECH LI-ION RS232 SERIAL COMMUNICATION CONSOLE DEBUG CABLE. Connect over my local network instead.

**Reply by alienatedsec** [reply]
@mrpp my battery has RS485 and this can transfer over the network using the same Wafeshare back to my laptop where the software can read its parameters the same way as it was directly connected. I am still working on HA integration for my battery using that connection. 

I am not sure how it works with Pylontech, but if it's RS232, you are likely to need other Waveshare module.

**Reply by mrpp** [reply]
i think my issue is the software is looking for com3 and i am trying to access tcp/ip?

**Reply by alienatedsec** [reply]
@mrpp You can have VirCom emulating your COM port on your laptop that will transfer from the Waveshare RS232 to the ETH device.

**Reply by mrpp** [reply]
![pylon vircom](https://github.com/user-attachments/assets/6693e350-d728-4e48-9717-547cc4100dfc)
Got this far but just won't communicate.

**Reply by alienatedsec** [reply]
@mrpp speed and other com port config in VirCom has to be the same as in your app

**Reply by mrpp** [reply]
> @mrpp speed and other com port config in VirCom has to be the same as in your app

yes when i use app i am using same 115200, like it's connecting just not talking. might switch the txd and rxd cables over.

**Reply by mrpp** [reply]
I am assuming set as a server, is transfer protocol ok like this with RS232, or should it be  Real_Com?   Thanks
![com1](https://github.com/user-attachments/assets/69b10af3-63c5-45c1-8e04-901b000ce4e1)

**Reply by alienatedsec** [reply]
@mrpp 
Is the interface on the battery `RS485` or `RS232`?

I am assuming (but doubt it) that the screenshot is the interface of the `RS232 to ETH` and not `RS485 to ETH` - if it is the latter, and the interface on the battery is `RS232`, it will not work because A and B connections for `RS485` are not equivalent to RX and TX for `RS232`.

Can you confirm the module you are using for your battery connection?
- https://thepihut.com/products/rs232-to-rj45-ethernet-module-1 
or 
- https://thepihut.com/products/industrial-serial-server-rs485-to-rj45-ethernet 

The VirCom Virtual Serial is either translating:
- from `RS485` via `RS485 to ETH` module to `RS232` COM1 on your PC or 
- from `RS232` via `RS232 to ETH` module to `RS232` COM1 on your PC.

According to this video https://www.youtube.com/watch?v=F-sj5_fNzKA , the console cable is `RS232`
![image](https://github.com/user-attachments/assets/9385694a-b0a2-4730-a8f9-efe17127a7a4)

**Reply by mrpp** [reply]
I have an Ethernet connection to battery with the pins wired to the RS232 to RJ45 Ethernet Module, I then have the main RJ45 from my poe hub to the waveshare. The Pylontech is RS232 with an RJ45 connector, so trying to communicate to the console port over my local network.
Yes RS232 to Ethernet being used.
Is it a case I need to convert back to RS232, in other words need another RS232 to Ethernet device, or should the data be read over the local network as it is now?

**Reply by alienatedsec** [reply]
The Ethernet port on the battery is not the Ethernet connection and it should be as follows.

From the Ethernet on the Battery should be wired (TX and RX) to RS232 (RX and TX) on your Waveshare, which is then connected to the network. Then you can access this port using VirCom as you already showcased above.

**Reply by mrpp** [reply]
yesthat is exactly as i have it. For some reason not communicating to my pc. 
![tx](https://github.com/user-attachments/assets/bc9a6b4b-f692-438a-9465-2699bfa92981)

**Reply by alienatedsec** [reply]
@mrpp have you tried anything else to make it working? Maybe the Waveshare module is faulty?

**Reply by mrpp** [reply]
Well, I'm starting to think maybe it is faulty. I can connect via USB with a Cisco cable so it's not the battery RS232 that's at fault.

**Reply by alienatedsec** [reply]
@mrpp what is the wiring after Cisco cable? Is it the same as on the wiring diagram you presented?

**Comment by alienatedsec** [comment]
@mrpp have you managed to make the connection in the end?

**Comment by mrpp** [comment]
No, for some reason it won't update. Sent from my HONOR phone-------- Original message --------From: alienatedsec ***@***.***>Date: Mon, 11 Nov 2024, 09:49To: alienatedsec/solis-ha-modbus-cloud ***@***.***>Cc: mrpp ***@***.***>, Mention ***@***.***>Subject: Re: [alienatedsec/solis-ha-modbus-cloud] Adding the RS232 Waveshare to acces Pylontech us5000 (Discussion #38)
@mrpp have you managed to make the connection in the end?

**Comment by mrpp** [comment]
Sorry for some reason it won't connect, not sure why, but think it's a bad rate issue? Sent from my HONOR phone-------- Original message --------From: alienatedsec ***@***.***>Date: Mon, 11 Nov 2024, 09:49To: alienatedsec/solis-ha-modbus-cloud ***@***.***>Cc: mrpp ***@***.***>, Mention ***@***.***>Subject: Re: [alienatedsec/solis-ha-modbus-cloud] Adding the RS232 Waveshare to acces Pylontech us5000 (Discussion #38)
@mrpp have you managed to make the connection in the end?
